mlir/lib/Dialect/SparseTensor/Transforms/Utils/SparseTensorLevel.h

```diff
@@ -21,42 +21,235 @@ class SparseTensorLevel {
   SparseTensorLevel &operator=(const SparseTensorLevel &) = delete;
```

Comment:
Can you please add a top level comment to the class SparseTensorLevel in this header (see all the other classes in headers in this Utils files). This is a rather central class for your code generation approach, so it is good to describe that this views the world from each dimension level, and how to use it